.github/workflows/build-publish.yml

```diff
@@ -0,0 +1,88 @@
+name: Build
+on: 
+  push:
+    branches:
+      - main
+    tags:
+    - 'v*'
+  pull_request:
+    branches:
+      - main
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NUGET_FEED: https://api.nuget.org/v3/index.json
+  NUGET_KEY: ${{ secrets.NUGET_API_KEY }}
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: Setup dotnet
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: '9.0.300'
```

Comment:
what is the best way to keep track of newer versions?

Reply:
It can be changed to '9.0' so it always uses the latest patch version

Reply:
@pmdevers I know. And `9.0.x` is also possible. The question is, if we thing about predictable builds, what is considered best these days?

Reply:
This is the way.

https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-net

Only discussion is if you want to pin the patch or minor version